src/MultiRegion/multi_region_boundary_conditions.jl

```diff
@@ -68,11 +68,11 @@ function fill_halo_regions!(c::MultiRegionObject, bcs, indices, loc, mrg::MultiR
     neighbors = Reference(c.regional_objects)
     buff      = Reference(buffers.regional_objects)
     
-    @apply_regionally event1 = fill_halo_event!(1, halo_tuple, c, indices, loc, arch, device_event(arch), mrg, neighbors, buff, args...; kwargs...)
-    @apply_regionally event2 = fill_halo_event!(2, halo_tuple, c, indices, loc, arch, event2,             mrg, neighbors, buff, args...; kwargs...)
-    @apply_regionally event3 = fill_halo_event!(3, halo_tuple, c, indices, loc, arch, event3,             mrg, neighbors, buff, args...; kwargs...)
+    event1 = construct_regionally(fill_halo_event!, 1, halo_tuple, c, indices, loc, arch, device_event(arch), mrg, neighbors, buff, args...; kwargs...)
+    event2 = construct_regionally(fill_halo_event!, 2, halo_tuple, c, indices, loc, arch, event2,             mrg, neighbors, buff, args...; kwargs...)
+    event3 = construct_regionally(fill_halo_event!, 3, halo_tuple, c, indices, loc, arch, event3,             mrg, neighbors, buff, args...; kwargs...)
```

Comment:
What do 1, 2, 3 stand for?

Reply:
Fill halo in direction 1, 2 and 3. We do not know which direction comes first a priori so the fill halo are anonymous. I could put it in a loop like we do in `fill_halo_regions.jl`.

Reply:
got it